src/core/main.c

```diff
@@ -2313,8 +2317,15 @@ static int initialize_runtime(
 
                         (void) os_release_status();
                         (void) hostname_setup(true);
-                        /* Force transient machine-id on first boot. */
-                        machine_id_setup(/* root= */ NULL, /* force_transient= */ first_boot, arg_machine_id, /* ret_machine_id */ NULL);
+
+                        /* Set up the machine ID flags */
+                        MachineIdSetupFlags machine_id_setup_flags = 0;
+                        if (arg_machine_id_from_firmware)
+                                machine_id_setup_flags = MACHINE_ID_SETUP_FORCE_FIRMWARE;
+                        else if (first_boot)
+                                machine_id_setup_flags = MACHINE_ID_SETUP_FORCE_TRANSIENT;
+
+                        machine_id_setup(/* root= */ NULL, arg_machine_id, /* flags= */ machine_id_setup_flags, /* ret_machine_id */ NULL);
```

Comment:
we only prefix parameters with `/* xyz= */` if what it is wasn't clear anyway from the parameter. i.e. for NULL paramaters and similar. but machine_id_setup_flags makes pretty clear what it is, hence drop `/* flags= */` here please